kong/conf_loader.lua

```diff
@@ -733,6 +734,33 @@ local function load(path, custom_conf)
         break
       end
     end
+
+    -- Validate origins
+    local seen_origins = {}
+    for i, v in ipairs(conf.origins) do
+      local from_scheme, from_host_port, to_host_port =
+        v:match("^(https?)://([^=]+:[%d]+)=https?://([^/]+)$")
+      if not from_scheme then
+        return nil, "origins must be of form 'from_scheme://from_host:from_port=to_scheme://to_host:to_port'"
+      end
+      -- Validate 'from'
+      local from_authority, err =
+        utils.format_host(utils.normalize_ip(from_host_port))
+      if not from_authority then
+        return nil, err
+      end
+      -- Check for duplicates
+      local from_origin = from_scheme .. "://" .. from_authority
+      if seen_origins[from_origin] then
+        return nil, "duplicate origin"
```

Comment:
This message is a bit unfriendly, isn't it?

1. It does not specify which origin is duplicated
2. If more than one origin is duplicated, the user will only find out after several trial and errors. Something that the `conf_loader` tries hard to avoid. Instead, it always tries to be a good citizen and report all errors at once, to avoid back and forth from the user.